frontend/packages/dev-console/src/components/import/image-search/ImageStream.tsx

```diff
@@ -1,192 +1,187 @@
     setFieldError,
-  ]);
-  const getImageStreamByNamespace = (ns: string) => {
-    const imageStreamsList = {};
-    const isBuilderImageNamespace = ns === builderImagesNamespace.Openshift;
-    const imgStreams = isBuilderImageNamespace
-      ? (normalizeBuilderImages(imageStreams) as NormalizedBuilderImages)
-      : (imageStreams as K8sResourceKind[]);
-    _.each(imgStreams, (img: BuilderImage | K8sResourceKind) => {
-      const { name, namespace } = isBuilderImageNamespace
-        ? (img as BuilderImage).obj.metadata
-        : (img as K8sResourceKind).metadata;
-      if (namespace === ns) {
-        imageStreamsList[name] = name;
-      }
-    });
-    return imageStreamsList;
-  };
-  const getImageStreamTags = (name: string) => {
-    const tags = {};
-    const imageStream = _.find(imageStreams, ['metadata.name', name]);
-    name &&
-      !_.isEmpty(imageStream) &&
-      _.each(imageStream.status.tags, ({ tag }) => {
-        tags[tag] = tag;
-      });
-    return tags;
-  };
-  React.useEffect(() => {
-    if (imageNamespace.value !== selectedProject) {
-      setselectedProject(imageNamespace.value);
-      setFieldValue('imageStream.image', '');
-      setFieldValue('imageStream.tag', '');
-      setFieldValue('isi', initialValues.isi);
-    }
-    if (imageName.value !== selectedName) {
-      setselectedName(imageName.value);
-      setFieldValue('imageStream.tag', '');
-    }
-    if (imageTag.value !== selectedTag) {
-      setselectedTag(imageTag.value);
-      imageTag.value !== '' && searchImageTag();
-    }
-  }, [
-    imageNamespace,
-    imageName,
-    imageTag,
-    selectedProject,
-    selectedName,
-    selectedTag,
-    setFieldValue,
-    initialValues.isi,
-    searchImageTag,
-  ]);
+    initialValues,
+  } = useFormikContext<FormikValues>();
+  const [hasAccessToPullImage, setHasAccessToPullImage] = React.useState(false);
+
+  const searchImageTag = React.useCallback(
+    (selectedTag: string) => {
+      setFieldValue('isSearchingForImage', true);
+      k8sGet(ImageStreamTagModel, `${imageStream.image}:${selectedTag}`, imageStream.namespace)
+        .then((imageStreamImport) => {
+          const { image, tag, status } = imageStreamImport;
+          const name = imageStream.image;
+          const isi = { name, image, tag, status };
+          const ports = getPorts(isi);
+          setFieldValue('isSearchingForImage', false);
+          setFieldValue('isi.name', name);
+          setFieldValue('isi.image', image);
+          setFieldValue('isi.tag', selectedTag);
+          setFieldValue('isi.ports', ports);
+          setFieldValue('image.ports', ports);
+          setFieldValue('name', getSuggestedName(name));
+          !application.name && setFieldValue('application.name', `${getSuggestedName(name)}-app`);
+          // set default port value
+          const targetPort = _.head(ports);
+          targetPort && setFieldValue('route.targetPort', makePortName(targetPort));
```

Comment:
Why are we setting first port as targetPort here. I think if we have a different port selected that should be used, if not then the first port should be used.

Reply:
@divyanshiGupta since we dont have option to select port here in deploy image flow, we are using the first as a default port.

Reply:
In routing section, if there are multiple ports available we can change the target port.

Reply:
My bad. The above piece of code is to set the default port for the first time, later if user wishes to change, they can do it.